File: yangxj96-common/src/main/java/io/github/yangxj96/common/base/BasicService.java
package io.github.yangxj96.common.base;

import com.baomidou.mybatisplus.extension.service.IService;

/**
 * RESTFul 接口公用service层
 *
 * @param <O> 子类对应的实体
 */
public interface BasicService<O extends BasicEntity> extends IService<O> {

    O create(O datum);

    boolean delete(String id);

    O modify(O datum);

}

File: yangxj96-common/src/main/java/io/github/yangxj96/common/base/BasicService.kt
package io.github.yangxj96.common.base

import com.baomidou.mybatisplus.extension.service.IService

/**
 * RESTFul 接口公用service层
 *
 * @param <O> 子类对应的实体
</O> */
interface BasicService<O : BasicEntity> : IService<O> {

    fun create(datum: O): O

    fun delete(id: String): Boolean

    fun modify(datum: O): O

}
